### ISSUE TITLE
config.js & .env

### ISSUE THREAD

**Issue by zeim839** [ufosc/Club_Website_2#2]
TODO: Set up [dotenv](https://www.npmjs.com/package/dotenv) and create config.js. 

[dotenv](https://www.npmjs.com/package/dotenv) is an environment variable reader/loader that essentially allows you to "store configuration in the environment separate from code." For example:
```
// My secret variables / my custom config:
SHA_SECRET = ABCDEF1234
MY_DB_PASSWORD = 123456
```
Instead of storing sensitive details as javascript variables in our code, we can create a `.env` file (as above). This way, anyone looking to run their own version of the website can supply their own `.env` without having to rewrite code.

That said, `.env` isn't exclusive to sensitive information only, It can be used to configure virtually any project variable (the "environment" of the project, hence its name). 

However, a problem arises when users don't supply `.env` files - either because they forgot or they don't want to modify the variable defaults. But since javascript is dynamically typed, they'll start receiving errors whenever any of these environment variables (which don't exist) are accessed. For this reason, we need a `config.js` file that sets our environment variables either to their .env file value or some other default. 

For now, you'll have to implement these variables:
```
SERVER_PORT
NODE_ENV
```

Once you've set up the [dotenv](https://www.npmjs.com/package/dotenv) package, create a `config.js` with the following (remove comments once done):
```js
// Read the NODE_ENV from .env (if it exists), otherwise set NODE_ENV to "development"
const NODE_ENV = process.env.NODE_ENV || "development"

// There's no difference between development/production at the moment 
// since we don't have many ENV variables.
const config = {
  "development": {
    port: process.env.SERVER_PORT || 3000
  },
  "production": {
    port: process.env.SERVER_PORT || 3000
  }
}

module.exports = config[env]
module.exports.ENV = NODE_ENV
```

Then, import the `config.js` file to `app.js`. Modify [this line](https://github.com/ufosc/Club_Website_2/blob/main/app.js#L9) to read the server port from the config module.

**Comment by Vinitvh** [comment]
Hi, I would like to work on this.

**Comment by Vinitvh** [comment]
@zeim839 Need some help modifying render function in app.js. Do I have to pass, `port` as an option beside page?

**Comment by zeim839** [comment]
No you dont, the code changed since Ive created the issue so the link isnt valid. You need to pass the port option on the last line of app.js, where it says http.listen().

**Comment by Vinitvh** [comment]
Okay. Got it.

**Comment by zeim839** [comment]
Resolved by #4 
Thanks @Vinitvh
